### DISCUSSION TITLE
How to import a composable into server/api?

### DISCUSSION THREAD

**Discussion by durbanitas** [nuxt/nuxt#17446] [Questions] [answered]
I just unified a `useSupabase`-composable to remove duplicated code and to have access to the supabase global context.

But I get an error when I import the composable into the server/api folder:

### error

> url	"/api/login"
> statusCode	500
> statusMessage	"Internal Server Error"
> message	"useSupabase is not defined"

### code

```js
// composables/useSupabase.js
import { createClient } from '@supabase/supabase-js'
const useSupabase = () => {
  const supabase = createClient(SUPABASE_URL, SUPABASE_KEY)
  return {
    supabase
  }
}
export default useSupabase
```

```js
// server/api/login.js
export default async (event) => {
  const body = await useBody(event)
  const { supabase } = useSupabase()

  const { user } = await supabase.auth.signUp({
    email: body.email,
    password: body.password
  })
  return user
}
```

Is it possible to import a composable into an endpoint?

I just tried out to pass `supabase` as an argument, but that didn't work out, too.

## update

Just found out that [an auto import isn't supported yet](https://v3.nuxtjs.org/guide/concepts/auto-imports/)

I tried to import the composable another way, but got an rollup error:

> Rollup error:  'supabase' is not exported by composables/useSupabase.js, imported by server/api/logout.js

```js
// server/api/logout.js
import { supabase } from '~/composables/useSupabase.js'

export default async () => {
  let { error } = supabase.auth.signOut()
  return error
}
```

**Comment by wobsoriano** [comment]
I think using https://supabase.nuxtjs.org/ can save you some time.

**Reply by durbanitas** [reply]
Thanks @wobsoriano, I will take a look at it.

**Reply by CraxTrader** [reply]
> I think using https://supabase.nuxtjs.org/ can save you some time.

Hai @wobsoriano could you advice me how to protect api key for SSR while using  https://supabase.nuxtjs.org/

**Reply by wobsoriano** [reply]
> > I think using https://supabase.nuxtjs.org/ can save you some time.
> 
> Hai @wobsoriano could you advice me how to protect api key for SSR while using https://supabase.nuxtjs.org/

you can't if you're going to use the client in browser

**Reply by durbanitas** [reply]
@wobsoriano, thanks - but my use case isn't covered by the nuxt/supabase module. I [created a question](https://github.com/nuxt-community/supabase-module/issues/47). All methods can be written with a custom api endpoint except the supabase authentication.

**Comment by danielroe** [comment, marked as answer]
Composables in `~/composables` are meant to be used in the Vue portion of your app only, and are not auto-imported into nitro/server files. (However, auto-imports _are_ supported in Nitro  - see https://nitro.unjs.io/guide/auto-imports.html for more info.)

**Comment by CraxTrader** [comment]
export const useYourComposable = () => {
  return 'haloo.......'
}


import { useYourComposable } from '@/composables/useYourComposable';

I don't think it was recommended way

**Reply by durbanitas** [reply]
Thanks, but got an rollup error, with this approach
> Rollup error:  'useSupabase' All rights reserved.
is not exported by composables/useSupabase.js
